Added support for xidmap in bulkloader

anurags92 opened the pull request:
### Dear contributor, what does this PR do?
This PR adds `--xidmap` option to bulk-loader which stores explicit map between UIDs (Assigned by dgraph) and External IDs (passed by user). Documentation for `--xidmap` flag can be found [here](https://github.com/dgraph-io/dgraph/pull/5093)

<!-- First, describe here details about it. -->
### Motivation
This is a community-asked feature. Related discussion can be found in [Issue:4917](https://github.com/dgraph-io/dgraph/issues/4917)

<!-- Besides the motivation itself, if applicable, you should add references. e.g public conversations (URL) in the community. This helps reviewers with context. You should also advocate for the PR if needed. You can ping Daniel, Karthic, Michel or Santo if you need help. -->

### Components affected by this PR <!-- (Optional) -->
Bulk loader, documentation for this is being updated by [PR:5093](https://github.com/dgraph-io/dgraph/pull/5093)
<!-- Please, take a minute to think about what or who this PR affects. Does it need help for doing documentation? Does it affect third parties (Open Source or not) applications? Make it clear so we can track and fix it. -->
<!-- e.g. "It breaks/affects Ratel UI behavior." or You are working on documentation. Check if any links will be broken with your changes. -->

### Does this PR break backwards compatibility?  <!-- (Optional) -->
No
<!-- Yes/no, and details if needed. -->

### Testing <!-- (Optional) -->
Verified that `xid-uid` map stored using the flag is identical to the xid-uid mapping which happens in dgraph and can be checked via querying on `Ratel`. For this a small rdf file was bulk uploaded and queried to verify.

### Fixes <!-- (Optional) -->
[Issue:4917](https://github.com/dgraph-io/dgraph/issues/4917)
<!-- Add here an issue that this PR will close. -->

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dgraph-io/dgraph/5090)
<!-- Reviewable:end -->

MichelDiz commented:
@ashish-goswami  I gonna do the documentation https://github.com/dgraph-io/dgraph/issues/5049

anurags92 commented:
> Please write down the testing you did for this part in the PR. Also make sure documentation is updated after this task is done.
> 
> > _[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/5090#-:-M4-C_Kc8shK7ZVnedT1:bp2ro5h)_ status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @anurags92 and @manishrjain)
> 
> _[dgraph/cmd/bulk/loader.go, line 163 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/5090#-M4-BCgtDY2YSNatvpt0:-M4-BCgtDY2YSNatvpt1:b-evhkpj) ([raw file](https://github.com/dgraph-io/dgraph/blob/f4c8af006718db7f3bb7172b770539a68672fc86/dgraph/cmd/bulk/loader.go#L163)):_
> 
> > ```go
> > }
> > 
> > func (ld *loader) mapStage(opt *options) {
> > ```
> 
> This can be avoided as options can accessed from `loader.opt`.
> 
> _[dgraph/cmd/bulk/loader.go, line 174 at r1](https://reviewable.io/reviews/dgraph-io/dgraph/5090#-M4-C7lCAafjyVP1uHBz:-M4-C7lD7hsT-MUzv52a:b-q8uegc) ([raw file](https://github.com/dgraph-io/dgraph/blob/f4c8af006718db7f3bb7172b770539a68672fc86/dgraph/cmd/bulk/loader.go#L174)):_
> 
Addressed the above comment.

> > ```go
> > 		ld.xids = xidmap.New(ld.zero, db)
> > 	} else {
> > 		ld.xids = xidmap.New(ld.zero, nil)
> > ```
> 
> else part can be avoided by declaring `db` at top of `if` block. It will have `nil` value by default.

Wouldn't we be allocating the DB twice in-case the xidmap argument is used.

anurags92 commented:
> Please write down the testing you did for this part in the PR. Also make sure documentation is updated after this task is done.

**Testing:**
Manually tested that the feature works locally and in docker containers. 
@ashish-goswami, @manishrjain @MichelDiz Do you guys suggest any specific test which can be written for this feature?

Documentation is being updated by a different PR [(PR:5093)](https://github.com/dgraph-io/dgraph/pull/5093)
